entrypoint.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+set -e
+
+
+#############################
+# Default Values
+
+# node
+: "${PRESTO_NODE_ENVIRONMENT:=docker}"
+: "${PRESTO_NODE_ID:=$(uuidgen)}"
+
+# config
+: "${PRESTO_CONF_COORDINATOR:=true}"
+: "${PRESTO_CONF_INCLUDE_COORDINATOR:=true}"
+: "${PRESTO_CONF_HTTP_PORT:=8080}"
+: "${PRESTO_CONF_DISCOVERY_SERVER_ENABLED:=true}"
+: "${PRESTO_CONF_DISCOVERY_URI:=http://localhost:8080}"
+: "${PRESTO_CONF_QUERY_MAX_MEMORY:=5GB}"
+: "${PRESTO_CONF_QUERY_MAX_MEMORY_PER_NODE:=1GB}"
+: "${PRESTO_CONF_QUERY_MAX_TOTAL_MEMORY_PER_NODE:=2GB}"
+
+
+# catalogs
+# jmx
+: "${PRESTO_CATALOG_JMX:=true}"
+: "${PRESTO_CATALOG_JMX_NAME:=jmx}"
+
+# hive
+: "${PRESTO_CATALOG_HIVE:=false}"
+: "${PRESTO_CATALOG_HIVE_NAME:=hive}"
+: "${PRESTO_CATALOG_HIVE_METASTORE_URI:=file}"
+: "${PRESTO_CATALOG_HIVE_RECURSIVE_DIRECTORIES:=true}"
+: "${PRESTO_CATALOG_HIVE_ALLOW_DROP_TABLE:=true}"
+: "${PRESTO_CATALOG_HIVE_USE_S3:=false}"
+: "${PRESTO_CATALOG_HIVE_S3_AWS_ACCESS_KEY:=}"
+: "${PRESTO_CATALOG_HIVE_S3_AWS_SECRET_KEY:=}"
+: "${PRESTO_CATALOG_HIVE_S3_ENDPOINT:=}"
+: "${PRESTO_CATALOG_HIVE_S3_USE_INSTANCE_CREDENTIALS:=false}"
+: "${PRESTO_CATALOG_HIVE_S3_SELECT_PUSHDOWN_ENABLED:=true}"
+
+
+#############################
+# node.properties
+
+echo "node.environment=${PRESTO_NODE_ENVIRONMENT}" >> /etc/presto/node.properties
+echo "node.id=${PRESTO_NODE_ID}" >> /etc/presto/node.properties
+echo "catalog.config-dir=/etc/presto/catalog" >> /etc/presto/node.properties
+echo "plugin.dir=/usr/lib/presto/lib/plugin" >> /etc/presto/node.properties
+
+
+#############################
+# config.properties
+
+echo "coordinator=${PRESTO_CONF_COORDINATOR}" >> /etc/presto/config.properties
+echo "node-scheduler.include-coordinator=${PRESTO_CONF_INCLUDE_COORDINATOR}" >> /etc/presto/config.properties
+echo "http-server.http.port=${PRESTO_CONF_HTTP_PORT}" >> /etc/presto/config.properties
+echo "discovery-server.enabled=${PRESTO_CONF_DISCOVERY_SERVER_ENABLED}" >> /etc/presto/config.properties
+echo "discovery.uri=${PRESTO_CONF_DISCOVERY_URI}" >> /etc/presto/config.properties
+echo "query.max-memory=${PRESTO_CONF_QUERY_MAX_MEMORY}" >> /etc/presto/config.properties
+echo "query.max-memory-per-node=${PRESTO_CONF_QUERY_MAX_MEMORY_PER_NODE}" >> /etc/presto/config.properties
+echo "query.max-total-memory-per-node=${PRESTO_CONF_QUERY_MAX_TOTAL_MEMORY_PER_NODE}" >> /etc/presto/config.properties
+
+
+#############################
+# catalogs
+
+# jmx
+if $PRESTO_CATALOG_JMX; then
+    echo "connector.name=jmx" >> "/etc/presto/catalog/${PRESTO_CATALOG_JMX_NAME}.properties"
+fi
+
+# hive
+if $PRESTO_CATALOG_HIVE; then
+    echo "connector.name=hive-hadoop2" >> "/etc/presto/catalog/${PRESTO_CATALOG_HIVE_NAME}.properties"
```

Comment:
I feel like there is probably a better way to all this appending, but I am no Bash wizard.  I'd be open to feedback here if anyone has better ideas.

Reply:
Because bash is wonderful, the better way of checking booleans looks like
```
if [ "$PRESTO_CATALOG_JMX" = true ];
```
https://stackoverflow.com/a/21210966/27130

Reply:
... that being said, this seems like a reasonable way to write out config files...

Reply:
But will this keep appending the same thing to the end of the file each time you do `docker run` without the `--rm`?  We might need a way of ensuring it only happens once to keep from littering.

Reply:
Yeah, it may actually do something weird like that.

At first I was using a heredoc, but it got really ugly because I couldn't get the indentation to look sensible.

I could have the first `echo` overwrite the file, then append after that?

Or I could keep looking into the heredoc and see if I can figure out the indentation.

Reply:
It seems I can do something like this:

```
{
    echo "foo"
    echo "bar"
} > /tmp/foobar
```
I'm going to give that a shot.

Reply:
Yep! That worked.. I'll push a comment to fix momentarily.